public/javascripts/gformFileUpload.js

```diff
@@ -5,543 +5,61 @@
-    return array.split(", ").map(function(a) {
-      if (humanReadableMimeTypes.hasOwnProperty(a))
-        return humanReadableMimeTypes[a];
-      return a; // default to mime type if no friendly value is found
+  function handleFileUpload(e) {
+    var file = e.target.files[0];
+    var form = $(e.target).closest("form")
+    var formGroup = form.find(".govuk-form-group")
+    var input = formGroup.find(".govuk-file-upload")
+    var submitButton = form.find(".govuk-button--secondary")
+    var formComponentId = input.attr("id");
+    var uploadedFiles = $("#" + formComponentId + "-files")
+    submitButton.css("display", "")
+    submitButton.on("click", function() {
+      formGroup.hide()
+      uploadedFiles.empty().append(startProgressBar());
+      submitButton.css("display", "none")
+      return true
     });
   }
 
-  function removeDuplicatesFromArray(array) {
-    return array.filter(function(item, index) {
-      return array.indexOf(item) >= index;
-    });
+  // Display uploading file message
+  function startProgressBar() {
+    return progressBarWrapper("<span class='app-progress-spinner'></span><span id='fileupload' role='alert'>" + strings.uploadingFile[lang] + "</span>", "");
   }
 
-  function transformMimeTypes(mimetypes) {
-    return removeDuplicatesFromArray(mapUserFriendlyFileTypes(mimetypes)).join(
-      ", "
+  function progressBarWrapper(messageContent, buttonContent) {
+    return $(
+      "<dl class='govuk-summary-list pp-file-upload-spinner__list'>" +
+        "<div class='govuk-summary-list__row'>" +
+          "<dd class='app-summary-list__spinner'>" +
+            messageContent +
+          "</dd>" +
+          "<dd class='app-summary-list__spinner'>" +
+            buttonContent +
+          "</dd>" +
+        "</div>" +
+      "</dl>"
     );
   }
 
-  function getFileExtension(fileName) {
-    var fa = fileName.split(".");
-    if (fa.length > 1) {
-      return fa.pop();
-    } else {
-      return "unknown";
-    }
-  }
-
-  function GformFileUpload() {
-    var self = this;
-
-    // Set up event handlers etc
-    function init() {
-      $(".govuk-file-upload").on("change", handleFileUpload);
-      $(".uploaded-files").on("click", ".govuk-link", handleFileDelete);
-    }
-
-    // Error handling
-    function handleError($input, msg) {
-      enableSubmitButton();
-      var errorEl =
-        '<span class="govuk-error-message" role="alert">' + msg + "</span>";
-      $(errorEl).insertBefore($input);
-    }
-
-    // Setup file upload
-    function handleFileUpload(e) {
-      var file = e.target.files[0];
-      if (!file) {
-        return false;
-      }
-      disableSubmitButton();
-      $(".govuk-error-message").remove();
-      var $input = $(e.currentTarget);
-      var dataset = $input[0].dataset;
-      var formTemplateId = dataset.formTemplateId;
-      var accessCode = dataset.accessCode;
-      var fileId = dataset.fileId;
-      var sectionNumber = dataset.sectionNumber;
```

Comment:
It seems like these attributes can be deleted https://github.com/hmrc/gform-frontend/blob/lxol/GFORMS-1780/app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala#L1099-L1105

Reply:
ok, looks like dataset is not being used anywhere else